spec/factories/document_factory.rb

```diff
@@ -4,13 +4,19 @@
   factory :document do
     content_id { SecureRandom.uuid }
     locale { I18n.available_locales.sample }
-    base_path { "/#{SecureRandom.alphanumeric(8)}" }
+    title { SecureRandom.alphanumeric(8) }
+    base_path { "#{DocumentTypeSchema.find(document_type).path_prefix}/#{title.parameterize}" }
     document_type { DocumentTypeSchema.all.reject(&:managed_elsewhere).sample.id }
 
     trait :with_body_in_schema do
       document_type do
         DocumentTypeSchema.all.select { |schema| schema.contents.any? { |field| field.id == "body" } }.sample.id
       end
     end
+
+    trait :without_title do
```

Comment:
This seems wrong as we're doing something and then negating it. I thought we were favouring a minimal set of defaults, which seems to have worked so far. Can we change it to :with_contents or something and lose the defaults, instead?